File: VB/Controllers/HomeController.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc
Imports DevExpress.Web.Mvc

Public Class HomeController
    Inherits Controller

    <AcceptVerbs(HttpVerbs.Get)> _
    Public Function Index() As ActionResult
        Return View(New Customer With {.ID = 0, .Name = "John", .Country = 2, .City = 2})
    End Function

    Public Function CountryPartial() As ActionResult
        Return PartialView(New Customer())
    End Function

    Public Function CityPartial() As ActionResult
        Dim country As Integer = If((Request.Params("Country") IsNot Nothing), Integer.Parse(Request.Params("Country")), -1)
        Return PartialView(New Customer With {.Country = country})
    End Function

    <AcceptVerbs(HttpVerbs.Post)> _
    Public Function Index(<ModelBinder(GetType(DevExpressEditorsBinder))> ByVal customer As Customer) As ActionResult
        If ModelState.IsValid Then
            'post customer to database
            Return RedirectToAction("Success")
        Else
            Return View(customer)
        End If
    End Function

    Public Function Success() As ActionResult
        Return View()
    End Function

End Class

File: CS/DevExpressMvc3CascadingCombo/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using DevExpress.Web.Mvc;
using CS.Models;
namespace CS.Controllers {
    public class HomeController : Controller {
        [AcceptVerbs(HttpVerbs.Get)]
        public ActionResult Index() {
            return View(new Customer { ID = 0, Country = 2, City = 2 });
        }
        public ActionResult CountryPartial() {
            return PartialView(new Customer());
        }
        public ActionResult CityPartial() {
            int country = (Request.Params["Country"] != null) ? int.Parse(Request.Params["Country"]) : -1;
            return PartialView(new Customer { Country = country });
        }
        [AcceptVerbs(HttpVerbs.Post)]
        public ActionResult Index([ModelBinder(typeof(DevExpressEditorsBinder))]Customer customer) {
            if(ModelState.IsValid) {
                //Post customer to database.
                return RedirectToAction("Success");
            }
            else
                return View(customer);
        }
        public ActionResult Success() {
            return View();
        }
    }
}